easypermissions/src/main/java/pub/devrel/easypermissions/helper/PermissionHelper.java

```diff
@@ -14,39 +13,38 @@
 /**
  * Delegate class to make permission calls based on the 'host' (Fragment, Activity, etc).
  */
-@RestrictTo(RestrictTo.Scope.LIBRARY)
 public abstract class PermissionHelper<T> {
 
-    private static final String TAG = "PermissionHelper";
-
     private T mHost;
 
     @NonNull
-    public static PermissionHelper newInstance(Activity host) {
+    public static PermissionHelper<Activity> newInstance(Activity host) {
         if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
-            return new LowApiPermissionsHelper(host);
+            return new LowApiPermissionsHelper<>(host);
         }
 
-        if (host instanceof AppCompatActivity) {
-            return new AppCompatActivityPermissionHelper((AppCompatActivity) host);
-        } else {
+        if (host instanceof AppCompatActivity)
+            //noinspection unchecked Hmmm... not sure what's going on here
+            return (PermissionHelper<Activity>) (Object)
```

Comment:
What's up with the double cast? `(PermissionHelper<Activity>) (Object)`

Reply:
Not sure, here's the error I get otherwise:
![untitled](https://user-images.githubusercontent.com/9490724/33635902-a932daee-d9ce-11e7-9f25-4e9499e61f53.png)

It works just fine at runtime so I'd guess we need to use some sort of type variance.

Reply:
Figured it out.